pontoon/sync/changeset.py

```diff
@@ -338,3 +367,53 @@ def bulk_update_translations(self):
                 'fuzzy',
                 'extra'
             ])
+            self.bulk_create_failing_checks(
+                Translation.objects.filter(pk__in=[t.pk for t in self.translations_to_update]).prefetch_related('entity')
+            )
+
+    def check_translation(self, resource, translations):
+        """
+        Return a list of failing checks. Empty list means that everything is okay with that translation.
+        """
+        CompareEntity = namedtuple(
+            'CompareEntity',
+            (
```

Comment:
cc @mathjazz @Pike Could you look if that covers all fields related to the Entities that should be passed to checks from compare locales? I'm trying to create a transport object with a visible interface. I'll move that in a better place after I'm done.